apps/storefront/src/app/[locale]/(main)/_components/navigation.tsx

```diff
@@ -1,62 +1,123 @@
+import { isValidJson } from "@/lib/helpers";
 import type { Maybe } from "@/lib/types";
 
 export const Navigation = ({ menu }: { menu: Maybe<Menu> }) => {
-  const router = useRouter();
+  // Close menu manually
+  const [currentMenuItem, setCurrentMenuItem] = useState("");
 
   if (!menu || menu?.items?.length === 0) {
     return null;
   }
 
   return (
-    <NavigationMenu className="mx-auto hidden pb-6 pt-3 md:flex">
-      <NavigationMenuList>
+    <NavigationMenu
+      onValueChange={setCurrentMenuItem}
+      value={currentMenuItem}
+      className="mx-auto hidden max-w-screen-xl pb-2 pt-2 md:flex"
+    >
+      <NavigationMenuList className="gap-6">
         {menu.items.map((item) => (
           <NavigationMenuItem key={item.id}>
-            <NavigationMenuTrigger
-              showIcon={!!item.children?.length}
-              onClick={() =>
-                item.category?.slug
-                  ? router.replace(
-                      paths.search.asPath({
-                        query: {
-                          category: item.category.slug,
-                        },
-                      }),
-                    )
-                  : undefined
-              }
+            <Link
+              href={item.url || "#"}
+              className="text-inherit no-underline hover:underline"
             >
-              {item.name}
-            </NavigationMenuTrigger>
+              <NavigationMenuTrigger showIcon={!!item.children?.length}>
+                {item.label}
+              </NavigationMenuTrigger>
+            </Link>
             {item.children?.length ? (
               <NavigationMenuContent>
-                <div className="grid w-[400px] p-2">
-                  {item.children?.map((child) => (
-                    <NavigationMenuLink asChild key={child.id} className="p-2">
-                      <Link href={generateLinkUrl(child, paths)}>
-                        <div className="text-sm font-medium leading-none group-hover:underline">
-                          {child.name ||
-                            child.collection?.name ||
-                            child.category?.name}
-                        </div>
-                      </Link>
-                    </NavigationMenuLink>
-                  ))}
+                <div className="grid w-full grid-cols-6 p-6">
+                  <div className="col-span-2 flex flex-col gap-3 pr-6">
+                    {item.children?.length
+                      ? item.children
+                          ?.filter((child) => !child.collectionImageUrl)
+                          .map((child) => (
+                            <Link
+                              key={child.id}
+                              href={child.url}
+                              className="group block space-y-1 rounded-md p-3 hover:bg-accent"
+                              onClick={() => {
+                                setCurrentMenuItem("");
+                              }}
```

Comment:
```suggestion
                              onClick={() => setCurrentMenuItem("")}
```
why the onClick event should set the component's state as `menu item = empty string`?

Reply:
It manually closes menu - if you click on given menu item menu should not be visible. It is workaround to achieve that.